*** LoginController.php	2016-12-11 00:59:32.372001531 +0000
--- ../LoginController.php	2016-12-11 00:49:57.274289203 +0000
***************
*** 213,218 ****
--- 213,227 ----
  		$originalUser = $user;
  		// TODO: Add all the insane error handling
  		/* @var $loginResult IUser */
+ 		try {		
+ 			$this->userManager->emit('\OC\User', 'preLoginValidation', array($user, $password));
+ 		} catch(\OC\User\LoginException $e) {
+ 			$this->session->set('loginMessages', [
+ 				['invalidpassword'], []
+ 			]);
+ 			$args = ['user' => $originalUser];
+                         return new RedirectResponse($this->urlGenerator->linkToRoute('core.login.showLoginForm', $args));
+ 		}
  		$loginResult = $this->userManager->checkPassword($user, $password);
  		if ($loginResult === false) {
  			$users = $this->userManager->getByEmail($user);
