--- original/apps/files_trashbin/lib/Storage.php	2018-10-06 14:38:24.000000000 +0200
+++ patched/apps/files_trashbin/lib/Storage.php	2018-10-06 14:40:16.000000000 +0200
@@ -182,7 +182,7 @@
 				$result = $this->doDelete($path, 'unlink', true);
 				unset(self::$moveOutOfSharedFolder[$this->mountPoint . $path]);
 			} else {
-				$result = $this->doDelete($path, 'unlink');
+				$result = $this->doDelete($path, 'unlink', true);
 			}
 		} catch (GenericEncryptionException $e) {
 			// in case of a encryption exception we delete the file right away
