diff -crB /original/nextcloud/lib/private/User/Backend.php /modified/nextcloud/lib/private/User/Backend.php
*** /original/nextcloud/lib/private/User/Backend.php	2017-12-27 17:37:23.316796734 +0100
--- /modified/nextcloud/lib/private/User/Backend.php	2017-12-27 20:30:59.828537353 +0100
***************
*** 48,54 ****
  
  	protected $possibleActions = array(
  		self::CREATE_USER => 'createUser',
- 		self::SET_PASSWORD => 'setPassword',
  		self::CHECK_PASSWORD => 'checkPassword',
  		self::GET_HOME => 'getHome',
  		self::GET_DISPLAYNAME => 'getDisplayName',
--- 48,53 ----
diff -crB /original/nextcloud/lib/private/User/Session.php /modified/nextcloud/lib/private/User/Session.php
*** /original/nextcloud/lib/private/User/Session.php	2017-12-27 17:37:23.316796734 +0100
--- /modified/nextcloud/lib/private/User/Session.php	2017-12-27 20:08:29.198591036 +0100
***************
*** 573,580 ****
  			// Ignore and use empty string instead
  		}
  
- 		$this->manager->emit('\OC\User', 'preLogin', array($uid, $password));
- 
  		$user = $this->manager->get($uid);
  		if (is_null($user)) {
  			// user does not exist
--- 573,578 ----
diff -crB /original/nextcloud/remote.php /modified/nextcloud/remote.php
*** /original/nextcloud/remote.php	2017-12-27 17:37:23.456795907 +0100
--- /modified/nextcloud/remote.php	2017-12-27 20:12:51.200322981 +0100
***************
*** 146,153 ****
  
  	// Load all required applications
  	\OC::$REQUESTEDAPP = $app;
! 	OC_App::loadApps(array('authentication'));
! 	OC_App::loadApps(array('filesystem', 'logging'));
  
  	switch ($app) {
  		case 'core':
--- 146,153 ----
  
  	// Load all required applications
  	\OC::$REQUESTEDAPP = $app;
! 	// Temporary fix because Amiv Cloud App won't be loaded otherwise.
! 	OC_App::loadApps();
  
  	switch ($app) {
  		case 'core':
