diff -crB /original/nextcloud/core/Controller/LoginController.php /modified/nextcloud/core/Controller/LoginController.php
*** /original/nextcloud/core/Controller/LoginController.php	2017-03-03 17:37:23.028798432 +0100
--- /modified/nextcloud/core/Controller/LoginController.php	2017-03-03 20:05:01.048718193 +0100
***************
*** 44,49 ****
--- 44,50 ----
  use OCP\IUser;
  use OCP\IUserManager;
  use OCP\IUserSession;
+ use OC\Hooks\PublicEmitter;
  
  class LoginController extends Controller {
  	/** @var IUserManager */
***************
*** 216,221 ****
--- 217,227 ----
  		}
  
  		$originalUser = $user;
+ 
+ 		if ($this->userManager instanceof PublicEmitter) {
+ 			$this->userManager->emit('\OC\User', 'preLogin', [$user, $password]);
+ 		}
+ 
  		// TODO: Add all the insane error handling
  		/* @var $loginResult IUser */
  		$loginResult = $this->userManager->checkPassword($user, $password);
diff -crB /original/nextcloud/lib/private/User/Backend.php /modified/nextcloud/lib/private/User/Backend.php
*** /original/nextcloud/lib/private/User/Backend.php	2017-03-03 17:37:23.316796734 +0100
--- /modified/nextcloud/lib/private/User/Backend.php	2017-03-03 20:30:59.828537353 +0100
***************
*** 48,54 ****
  
  	protected $possibleActions = array(
  		self::CREATE_USER => 'createUser',
- 		self::SET_PASSWORD => 'setPassword',
  		self::CHECK_PASSWORD => 'checkPassword',
  		self::GET_HOME => 'getHome',
  		self::GET_DISPLAYNAME => 'getDisplayName',
--- 48,53 ----
diff -crB /original/nextcloud/lib/private/User/Session.php /modified/nextcloud/lib/private/User/Session.php
*** /original/nextcloud/lib/private/User/Session.php	2017-03-03 17:37:23.316796734 +0100
--- /modified/nextcloud/lib/private/User/Session.php	2017-03-03 20:08:29.198591036 +0100
***************
*** 318,323 ****
--- 318,327 ----
  								IRequest $request,
  								OC\Security\Bruteforce\Throttler $throttler) {
  		$currentDelay = $throttler->sleepDelay($request->getRemoteAddress());
+ 		
+ 		if ($this->manager instanceof PublicEmitter) {
+ 			$this->manager->emit('\OC\User', 'preLogin', [$user, $password]);
+ 		}
  
  		$isTokenPassword = $this->isTokenPassword($password);
  		if (!$isTokenPassword && $this->isTokenAuthEnforced()) {
***************
*** 463,469 ****
  	 * @throws LoginException if an app canceld the login process or the user is not enabled
  	 */
  	private function loginWithPassword($uid, $password) {
- 		$this->manager->emit('\OC\User', 'preLogin', array($uid, $password));
  		$user = $this->manager->checkPassword($uid, $password);
  		if ($user === false) {
  			// Password check failed
--- 467,472 ----
***************
*** 513,520 ****
  			// Ignore and use empty string instead
  		}
  
- 		$this->manager->emit('\OC\User', 'preLogin', array($uid, $password));
- 
  		$user = $this->manager->get($uid);
  		if (is_null($user)) {
  			// user does not exist
--- 516,521 ----
diff -crB /original/nextcloud/remote.php /modified/nextcloud/remote.php
*** /original/nextcloud/remote.php	2017-03-03 17:37:23.456795907 +0100
--- /modified/nextcloud/remote.php	2017-03-03 20:12:51.200322981 +0100
***************
*** 146,153 ****
  
  	// Load all required applications
  	\OC::$REQUESTEDAPP = $app;
! 	OC_App::loadApps(array('authentication'));
! 	OC_App::loadApps(array('filesystem', 'logging'));
  
  	switch ($app) {
  		case 'core':
--- 146,153 ----
  
  	// Load all required applications
  	\OC::$REQUESTEDAPP = $app;
! 	// Temporary fix because Amiv Cloud App won't be loaded otherwise.
! 	OC_App::loadApps();
  
  	switch ($app) {
  		case 'core':
